projects/packages/connection/src/class-connection-notice.php

```diff
@@ -0,0 +1,223 @@
+		$deleting_connection_owner = in_array( $connection_owner_id, (array) $user_ids_to_delete, true );
+		if ( ! $deleting_connection_owner ) {
+			return;
+		}
+
+		// Bail if they're trying to delete themselves to avoid confusion.
+		if ( get_current_user_id() === $connection_owner_id ) {
+			return;
+		}
+
+		$tracking = new Tracking();
+
+		// Track it!
+		if ( method_exists( $tracking, 'record_user_event' ) ) {
+			$tracking->record_user_event( 'delete_connection_owner_notice_view' );
+		}
+
+		$connected_admins = $connection_manager->get_connected_users( 'jetpack_disconnect' );
+		$user             = is_a( $connection_owner_userdata, 'WP_User' ) ? esc_html( $connection_owner_userdata->data->user_login ) : '';
+
+		echo "<div class='notice notice-warning' id='jetpack-notice-switch-connection-owner'>";
+		echo '<h2>' . esc_html__( 'Important notice about your Jetpack connection:', 'jetpack-connection' ) . '</h2>';
+		echo '<p>' . sprintf(
+			/* translators: WordPress User, if available. */
+			esc_html__( 'Warning! You are about to delete the Jetpack connection owner (%s) for this site, which may cause some of your Jetpack features to stop working.', 'jetpack-connection' ),
+			esc_html( $user )
+		) . '</p>';
+
+		if ( ! empty( $connected_admins ) && count( $connected_admins ) > 1 ) {
+			echo '<form id="jp-switch-connection-owner" action="" method="post">';
+			echo "<label for='owner'>" . esc_html__( 'You can choose to transfer connection ownership to one of these already-connected admins:', 'jetpack-connection' ) . ' </label>';
+
+			$connected_admin_ids = array_map(
+				function ( $connected_admin ) {
+					return $connected_admin->ID;
+				},
+				$connected_admins
+			);
+
+			wp_dropdown_users(
+				array(
+					'name'    => 'owner',
+					'include' => array_diff( $connected_admin_ids, array( $connection_owner_id ) ),
+					'show'    => 'display_name_with_login',
+				)
+			);
+
+			echo '<p>';
+			submit_button( esc_html__( 'Set new connection owner', 'jetpack-connection' ), 'primary', 'jp-switch-connection-owner-submit', false );
+			echo '</p>';
+
+			echo "<div id='jp-switch-user-results'></div>";
+			echo '</form>';
+			?>
+			<script type="text/javascript">
+				jQuery( document ).ready( function( $ ) {
+					$( '#jp-switch-connection-owner' ).on( 'submit', function( e ) {
+						var formData = $( this ).serialize();
+						var submitBtn = document.getElementById( 'jp-switch-connection-owner-submit' );
+						var results = document.getElementById( 'jp-switch-user-results' );
+
+						submitBtn.disabled = true;
+
+						$.ajax( {
+							type        : "POST",
+							url         : "<?php echo esc_url( get_rest_url() . 'jetpack/v4/connection/owner' ); ?>",
+							data        : formData,
+							headers     : {
+								'X-WP-Nonce': "<?php echo esc_js( wp_create_nonce( 'wp_rest' ) ); ?>",
+							},
+							success: function() {
+								results.innerHTML = "<?php esc_html_e( 'Success!', 'jetpack-connection' ); ?>";
```

Comment:
As above, probably ok (here we _want_ the entity encoding, and it's unlikely anyone will put `\` in a translation) but to really be correct it should pass it through `wp_json_encode` too to ensure it's valid JS.
```suggestion
								results.innerHTML = "<?php echo wp_json_encode( esc_html__( 'Success!', 'jetpack-connection' ), JSON_HEX_TAG | JSON_HEX_AMP ); ?>";
```

Reply:
Done: https://github.com/Automattic/jetpack/pull/29087/commits/b9e5d2764affd27fe12a3d7ed0d31f850a84f4a4